src/app/app.lastSearchesService.js

```diff
@@ -0,0 +1,29 @@
+export default class LastSearchesService {
+    constructor(lastSearches) {
+        this.lastSearches = lastSearches;
+    }
+
+    saveCity(city) {
+        if (city === undefined || city.trim() === '') {
+            return;
+        } else {
```

Comment:
If you are returning in the if block you don't have to wrap your logic in the else block